keyboards/converter/usb_usb/custom_matrix.cpp

```diff
@@ -39,6 +39,66 @@ extern "C" {
 #include "quantum.h"
 }
 
+/* see keymaps/via/keymap.c for VIA implementation details */
+#ifdef VIA_ENABLE
```

Comment:
why is there via specific implementation here?

Reply:
That doesn't really answer the question, though.  Why does there need to be via specific handling of this? 

VIA should be unaware of the hardware, here.   The dynamic keymap feature, which via leverages changes the keymap, and that reads from the matrix config.

Eg, there is not really a good reason to have via specific code if the matrix is properly implemented, as far as I can see.

Reply:
I'm still not 100% sure I understand the issue, but is it that you'd prefer a different ifdef to wrap that code?

There are two scan matrices (the original, which supports translation of all 256 USB keycodes but only one layer, and the new one I've added, which supports fewer keycodes in order to allow for three layers).  Both matrices are implemented correctly, and the USB-to-USB converter can support both - but you have to choose one.  Since only one of the two matrices supports multiple layers, it made sense to me to put it under an ifdef so that you can make that compile-time choice.  If the issue you're getting at is that using VIA_ENABLE is too specific a toggle in this hardware-specific code to make that choice (since any dynamic keymap user might want to enable it) then I could put the smaller one under some other ifdef and then enable that new ifdef only in the via keymap, which would take the VIA ifdef out of custom_matrix.cpp.